### ISSUE TITLE
systemd-resolved on Ubuntu 18.04 rejects NetworkManager's Tailscale DNS config

### ISSUE THREAD

**Issue by danderson** [tailscale/tailscale#1879]
Found while debugging something else:

```
May 06 22:13:29 dave-VirtualBox NetworkManager[9221]: <warn>  [1620364409.9112] dns-sd-resolved[0x562a8dec9620]: Failed: GDBus.Error:System.Error.E2BIG: Argument list too long
```

Setting our extremely long split-DNS domain routing list fails, because apparently old D-Bus had a low-ish limit on argument count, I guess. So, NetworkManager successfully sets our DNS server, but not any search or routing domains, so the split DNS config never gets used.

As usual, my preferred fix is "upgrade to 20.04, the new LTS". Another option would be to remove the in-arpa searches from our list, which would drastically shrink it - at the cost of breaking reverse lookups. Ideally we'd do this only for versions of the software that reject our configs, but of course NM happily accepts our config, and mostly ignores the failure to push into resolved.

**Comment by cg31** [comment]
Sometimes upgrading to 20.04 is not an option, for exampple, Microsoft Azuere only has 18.04 offcially now.

**Comment by danderson** [comment]
The problem is that there's not a huge amount we can do about this kind of thing without sacrificing some functionality in our code. In this case, breaking reverse lookups is the obvious option.

We would still need to detect that this has happened, so we can try to compensate. Unfortunately this software stack doesn't make that easy.

In the meantime, setting a non-split DNS config still works fine, since it doesn't require a huge domain list to capture reverse lookups.

**Comment by Xe** [comment]
This happens on Ubuntu 20.04 on Unstable too:

![20210607_07h13m25s_grim](https://user-images.githubusercontent.com/529003/121012278-8dd36d80-c765-11eb-8a68-4f37438d8483.png)

Magic DNS lookups fail (shachi is a host in my tailnet):

![20210607_07h17m18s_grim](https://user-images.githubusercontent.com/529003/121012277-8dd36d80-c765-11eb-8ac9-d8afb0d7bc1e.png)

Tailscale selected NetworkManager as its DNS provider:

![20210607_07h24m22s_grim](https://user-images.githubusercontent.com/529003/121012276-8dd36d80-c765-11eb-93fe-be19494949d3.png)

NetworkManager says no:

![20210607_07h31m40s_grim](https://user-images.githubusercontent.com/529003/121012275-8dd36d80-c765-11eb-9207-20a6f11ba3e3.png)

systemd-resolved isn't getting the Magic DNS config:

![20210607_07h46m38s_grim](https://user-images.githubusercontent.com/529003/121012274-8dd36d80-c765-11eb-8521-f05466f6924e.png)

Normally the resolved output would look something like this (note: this may not be the most analogous case as the machine i'm getting the output on is not using networkmanager but either way you can see that the output is very different and doesn't include the quad 100 address):

```
Link 12 (tailscale0)
    Current Scopes: DNS
         Protocols: +DefaultRoute -LLMNR -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: 100.100.100.100
       DNS Servers: 100.100.100.100
        DNS Domain: akua akua.xeserv.us christine.website.beta.tailscale.net localhost ~.
```

But somehow things just aren't being plumbed through to resolved, which seems to be the ultimate arbiter of DNS on Ubuntu 20.04, at least on the desktop.

**Comment by Xe** [comment]
Interestingly this does work on Ubuntu 20.04 Server and the cloud image until NetworkManager is installed. Reproduction steps:

1. Install a new Ubuntu Server VM from the upstream ISO/cloud image
2. Install tailscale
3. Activate tailscale
3. Try to use Magic DNS (it should work)
4. Install NetworkManager `sudo apt install network-manager`
5. reboot
6. Try to use Magic DNS (it should fail)

**Comment by danderson** [comment]
Fixed indirectly, via #1945. 18.04's NetworkManager is so old that it does broadly the wrong thing anyway, so the new logic bypasses it and achieves DNS supremacy some other way.
